File: apis/src/main/java/com/beat/apis/performance/application/dto/modify/schedule/ScheduleModifyResponse.java
package com.beat.apis.performance.application.dto.modify.schedule;

import java.time.LocalDateTime;

import com.beat.apis.schedule.application.dto.ScheduleNumberType;

public record ScheduleModifyResponse(
	Long scheduleId,
	LocalDateTime performanceDate,
	int totalTicketCount,
	int dueDate,
	ScheduleNumberType scheduleNumber
) {
	public static ScheduleModifyResponse of(Long scheduleId, LocalDateTime performanceDate, int totalTicketCount,
		int dueDate, ScheduleNumberType scheduleNumber) {
		return new ScheduleModifyResponse(scheduleId, performanceDate, totalTicketCount, dueDate, scheduleNumber);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/ScheduleModifyResponse.kt
package com.beat.apis.performance.api.response

import com.beat.apis.schedule.api.type.ScheduleNumberType
import com.beat.apis.performance.application.result.ScheduleResult
import java.time.LocalDateTime

@ConsistentCopyVisibility
data class ScheduleModifyResponse private constructor(
    val scheduleId: Long?,
    val performanceDate: LocalDateTime?,
    val totalTicketCount: Int,
    val dueDate: Int,
    val scheduleNumber: ScheduleNumberType?,
) {
    companion object {
        fun from(result: ScheduleResult): ScheduleModifyResponse = ScheduleModifyResponse(
            scheduleId = result.id,
            performanceDate = result.performanceDate,
            totalTicketCount = result.totalTicketCount,
            dueDate = result.dueDate,
            scheduleNumber = result.scheduleNumber?.let(ScheduleNumberType::valueOf),
        )
    }
}
